src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml

```diff
@@ -2834,13 +2851,25 @@ module Ledger = struct
     in
     apply_parties_transaction l txn (Js.to_string account_creation_fee)
 
+  let create_token_account pk token =
+    Mina_base.Account_id.create (public_key pk) (token_id token)
+    |> Mina_base.Account_id.public_key
+    |> Signature_lib.Public_key.Compressed.to_string |> Js.string
+
+  let custom_token_id pk token =
+    Mina_base.Account_id.derive_token_id
+      ~owner:(Mina_base.Account_id.create (public_key pk) (token_id token))
+    |> Mina_base.Token_id.to_string |> Js.string
```

Comment:
I think we need to cover the case where `pk` and `token` represent variables, see comments I'll add on the other PR. possibly cover that in a separate function `custom_token_id_checked`. so that both can have the same interface `(pk: PublicKey, token: Field) => Field`, I think probably this should return a `Field` as well

Reply:
Got it! Added support for a `Checked` version, mind taking a look if the new implementation is correct?